src/main/java/com/iota/iri/service/milestone/MilestoneSolidifier.java

```diff
@@ -146,118 +145,191 @@ public void shutdown() {
     }
 
     /**
-     * This method contains the logic for the milestone solidification, that gets executed in a separate {@link Thread}.
+     * This method takes an entry from the {@link #unsolidMilestonesPool} and adds it to the
+     * {@link #milestonesToSolidify} queue.
      *
-     * It periodically updates and checks the unsolid milestones by invoking {@link #processSolidificationTask()}.
+     * It first checks if the given milestone is already part of the queue and then tries to add it. If the queue is not
+     * full yet, the addition to the queue is relatively cheap, because the {@link #youngestMilestoneInQueue} marker can
+     * be updated without iterating over all entries. If the queue reached its capacity already, we replace the entry
+     * marked by the {@link #youngestMilestoneInQueue} marker and update the marker by recalculating it using
+     * {@link #determineYoungestMilestoneInQueue()}.
      *
-     * To allow for faster processing it only "waits" for another check if the current solidification task was not
-     * finished (otherwise it immediately "continues" with the next one).
+     * @param milestoneEntry entry from the {@link #unsolidMilestonesPool} that shall get added to the queue
      */
-    private void milestoneSolidificationThread() {
-        while(!Thread.interrupted()) {
-            if (processSolidificationTask()) {
-                continue;
+    private void addToSolidificationQueue(Map.Entry<Hash, Integer> milestoneEntry) {
+        if (milestonesToSolidify.containsKey(milestoneEntry.getKey())) {
+            return;
+        }
+
+        if (milestonesToSolidify.size() < SOLIDIFICATION_QUEUE_SIZE) {
+            milestonesToSolidify.put(milestoneEntry.getKey(), milestoneEntry.getValue());
+
+            if (youngestMilestoneInQueue == null || milestoneEntry.getValue() > youngestMilestoneInQueue.getValue()) {
+                youngestMilestoneInQueue = milestoneEntry;
             }
+        } else if (milestoneEntry.getValue() < youngestMilestoneInQueue.getValue()) {
+            milestonesToSolidify.remove(youngestMilestoneInQueue.getKey());
+            milestonesToSolidify.put(milestoneEntry.getKey(), milestoneEntry.getValue());
 
-            ThreadUtils.sleep(SOLIDIFICATION_INTERVAL);
+            youngestMilestoneInQueue = null;
```

Comment:
You already have `youngestMilestoneInQueue = null;` as part of `determineYoungestMilestoneInQueue`

Reply:
true